boards/DIM/Test/Src/Test_StateMachine.cpp

```diff
@@ -79,6 +97,10 @@ class DimStateMachineTest : public SevenSegDisplaysTest,
         RegenPaddleTest::TearDown();
         RotarySwitchTest::TearDown();
 
+        BinarySwitchTest::TearDownBinarySwitch(start_switch);
```

Comment:
Instead of writing 
```
ASSERT(foo != NULL);
App_Foo_Destroy(foo);
foo = NULL;
```
I thought it would be more convenient and less error-prone to create a function specifically for this: `TearDownBinarySwitch`. This means each test module (e.g. `BinarySwitchTest`, `RotarySwitchTest`) needs to provide a teardown function (e.g. `TearDownBinarySwitch`, `TearDownRotarySwitch`). What do you think of this? Maybe we could even get away with a macro that works for every module?

```
#define teardown_object(module, instance_ptr) \
  ASSERT(instance_ptr != NULL); \
  App_##module##_Destroy(instance_ptr); \
  instance_ptr = NULL
```

Reply:
Did something similar for `Led` in #589 btw

Reply:
I think either option is fine here. I think I'd lean towards the function instead of the macro because it's a bit more explicit and comprehensible.